cores/esp8266/Esp.cpp

```diff
@@ -664,10 +664,16 @@ bool EspClass::flashEraseSector(uint32_t sector) {
 }
 
 #if PUYA_SUPPORT
+#if PUYA_BUFFER_SIZE % 256 != 0
+#error PUYA_BUFFER_SIZE is not 256 byte aligned
+#endif
 static SpiFlashOpResult spi_flash_write_puya(uint32_t offset, uint32_t *data, size_t size) {
     if (data == nullptr) {
       return SPI_FLASH_RESULT_ERR;
     }
+    if (size % 4 != 0) {
```

Comment:
Shouldn't 4-byte alignment also be enforced on offset here, given that the corner cases are handled above this function?

Reply:
This is an internal function, it's only used by flashWrite, but I can add the restrictions